### PR TITLE
Upgrade dependencies for Stylelint v16 compat

### PR DESCRIPTION
**Pull request by ProchaLu**
This PR fixes the compatibility issues caused by the outdated Stylelint dependencies. To allow smooth functionality and compatibility with Preflight, we will update the Stylelint dependencies to the latest stable versions. Additionally, we added a Stylelint test to the GitHub workflow, which allows us to detect a Stylelint error earlier.

```
TypeError: Cannot read properties of undefined (reading 'rejected')
    at Object.<anonymous> (/Users/lukas/Documents/UpLeveled/preflight/stylelint-config-upleveled/node_modules/.pnpm/stylelint-scss@5.3.0_stylelint@16.0.2/node_modules/stylelint-scss/src/rules/comment-no-empty/index.js:12:42)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/lukas/Documents/UpLeveled/preflight/stylelint-config-upleveled/node_modules/.pnpm/stylelint-scss@5.3.0_stylelint@16.0.2/node_modules/stylelint-scss/src/rules/index.js:31:23)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,21 +23,23 @@
...
+    "stylelint-config-recommended": "14.0.0",
+    "stylelint-config-recommended-scss": "14.0.0"
   },
   "devDependencies": {
     "@types/prettier": "2.7.3",
     "eslint": "8.55.0",
     "eslint-config-upleveled": "7.3.4",
-    "stylelint": "15.11.0",
+    "stylelint": "16.0.2",
     "stylelint-config-upleveled": "1.0.3",
```
Comment (karlhorky) [inline package.json:37]: can we remove this? it was only used with the config for the tests, right? and now we config using `.`

Comment (ProchaLu) [reply package.json:39]: removed 
[8ded186](https://github.com/upleveled/stylelint-config-upleveled/pull/56/commits/8ded186f314a117e201de4a0320510f3e0c021cf)

File: __tests__/stylelint.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { $ } from 'execa';
+import { expect, test } from 'vitest';
+
+test('run stylelint and expect errors', async () => {
```
Comment (karlhorky) [inline __tests__/stylelint.test.ts:4]:
```suggestion
test('Errors correctly reported with Stylelint config', async () => {
```

File: package.json [hunk 3]
```diff
@@ -22,22 +22,26 @@
...
-    "stylelint-config-recommended-scss": "13.1.0"
+    "stylelint-config-recommended": "14.0.0",
+    "stylelint-config-recommended-scss": "14.0.0"
   },
   "devDependencies": {
     "@types/prettier": "2.7.3",
     "eslint": "8.55.0",
     "eslint-config-upleveled": "7.3.4",
-    "stylelint": "15.11.0",
     "stylelint-config-upleveled": "1.0.3",
```
Comment (karlhorky) [inline package.json:39]: can we remove this? it was only used with the config for the tests, right? and now we config using `.`

**Review by karlhorky — APPROVED** [review]

**Review by karlhorky — APPROVED** [review]

**Comment by karlhorky** [comment]
Released in `stylelint-config-upleveled@1.0.4`:

- https://github.com/upleveled/stylelint-config-upleveled/releases/tag/v1.0.4

**Pull request merged**